src/parser/binary-operation-parser.ts

```diff
@@ -30,17 +30,16 @@ import { ValueNode } from '../operation-node/value-node.js'
 import { WhereInterface } from '../query-builder/where-interface.js'
 import { HavingInterface } from '../query-builder/having-interface.js'
 import { createJoinBuilder, createSelectQueryBuilder } from './parse-utils.js'
-import { ExpressionOrFactory } from './expression-parser.js'
 import { OperationNode } from '../operation-node/operation-node.js'
 import { Expression } from '../expression/expression.js'
 
-export type FilterValueExpression<
+export type OperandValueExpression<
```

Comment:
I think this should still be `FilerValueExpression`. It's still designed just for filter methods like `where`, `having` and `on`